Feat/read only offline access (#110)
pod-os/PodOS: docs/decisions/0011-read-only-offline-access.md

  * The rdflib Updater would also need to be modified to support offline writes.
  * Slightly more complex initial implementation.
  * Still needs decisions about storage format and cache eviction policy.
  * Only works when using the rdflib fetcher, not when using fetch directly.
  * Only covers RDF documents, not other resources (e.g., images, files).
  * Does not run in service worker and thus cannot sync after the app is closed.

**Considered.** Provides a scalable, modular, and future-proof architecture.

### Service Worker-Based Cache (HTTP Caching of RDF Files)

**Approach:** Use a Service Worker to intercept HTTP fetches and return cached RDF documents when offline, using the browser’s `CacheStorage`.

* **Pros:**

  * Easy to set up with Workbox or native API.
  * Can cache any HTTP resource, not just RDF documents.
  * Works with any fetch call, not just rdflib.
  * Offline-writes could use a [keep-revert conflict resolution](https://remotestoragejs.readthedocs.io/en/v1.2.3/contributing/internals/cache-data-format.html#keep-revert-conflict-resolution]).
  * Can do background sync for offline writes.

* **Cons:**

>   * Lacks insight into rdflib or RDF document structure, RDF-graph based merging strategies might be hard to implement later.

Reviewer: ```suggestion
  * Lacks insight into rdflib or RDF document structure, so RDF-graph based merging strategies would need completely separate implementation later.
```

>   * PATCH requests with n3/patch or sparql/update could be intercepted, but cannot update the cache while offline

Reviewer: ```suggestion
  * By design would pass through PATCH requests as this solution has no natural mechanism to inspect and process n3/patch or sparql/update. Offline support would need to come from a separate cache layer.
```

    * This problem might be mitigated by the fact, that rdflib.js updates the store after a (seemingly) successful request.
    * Yet if refreshing the page, the rdflib store gets lost and the cache is not up-to-date.
  * Is based on browser's service worker and will not work in Node.js out of the box.
  * Offline cache needs to store different serialization formats (e.g., Turtle, RDF/XML, JSON-LD).

> **Rejected** The limitations in handling offline PATCH requests and maintaining cache consistency make this approach less suitable for future offline write support.

Reviewer: ```suggestion
**Rejected** The need for a separate cache to handle offline PATCH requests and maintain cache consistency makes this approach less suitable for future offline write support.
```
